### PR TITLE
Added option for setting gss mech oid (allows to implement SPNEGO authentication)

### PR DESCRIPTION
**Pull request by veklov**
@02strich - Hello, I have ported https://github.com/apple/ccs-pykerberos/issues/47. It adds option for setting gss mech oid. If you agree and merge in this change, I am going to use this feature to implement SPNEGO authentication in requests_kerberos and urllib_kerberos. Here are corresponding forks:
https://github.com/mongodb-labs/winkerberos/pull/15
https://github.com/veklov/requests-kerberos
https://github.com/veklov/urllib_kerberos

### CODE REVIEW

File: src/kerberos.c [hunk 1]
```diff
@@ -120,18 +126,31 @@ static PyObject* authGSSClientInit(PyObject* self, PyObject* args, PyObject* key
...
     }
 
+    if (pymech_oid != NULL) {
+        if (!PyCheck(pymech_oid)) {
+            PyErr_SetString(PyExc_TypeError, "Invalid type for mech_oid");
+            return NULL;
+        }
+        mech_oid = PyGet(pymech_oid, gss_OID_desc);
+        if (mech_oid == NULL) {
+            return NULL;
```
Comment (02strich) [inline src/kerberos.c]: is there an error message missing here?

Comment (veklov) [inline src/kerberos.c]: fixed

**Comment by 02strich** [comment]
Thank you for the contribution. I have merged it and published 1.14 to PyPi

**Comment by veklov** [comment]
That was quick. Thank you!

**Pull request merged**